### PR TITLE
Add test for DS_configAdmin Operation Order

### PR DESCRIPTION
**Pull request by aadityap-mathworks**
Changes:
- Added testModificationBeforeDeactivation testpoint to verify that DS must complete modification before deactivation of a component
- Added testModificationOrder testpoint to verify that DS must complete modification before another modification can be applied

Signed-off-by: The MathWorks, Inc. <aadityap@mathworks.com>

### CODE REVIEW

File: compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  ASSERT_TRUE(service) << "GetService failed for CAInterface";
+
+  compConfigs = GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  ASSERT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
+    << "Component state should be ACTIVE";
+
+  const std::string update1{ "update1" };
+  props["uniqueProp"] = update1;
+  configObject->Update(props);
```
Comment (Jeeyuen-Rick) [inline compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp:80]: See Patty's changes from "update methods to return future"
auto fut = configuration->Update(props);
fut.get();

File: compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "TestFixture.hpp"
+#include "gtest/gtest.h"
+
+#include "cppmicroservices/Constants.h"
+#include "cppmicroservices/ServiceObjects.h"
```
Comment (Jeeyuen-Rick) [inline compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp:27]: Do we need these headers?

Comment (apatil19) [reply compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp:27]: Nope. Took those out.

File: compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp (hunk 1 above)
Comment (apatil19) [reply compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp:80]: I'm not doing anything with the returned future; in this test case, simply waiting and deactivating bundle in next line.
and validating that even though I'm not waiting on the future, DS does the modification first and Deactivation after that.

File: compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  // Create configuration object
+  auto configObject = configAdminService->GetConfiguration(componentName);
+
+  // Update property
+  cppmicroservices::AnyMap props(
+    cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  const std::string instanceId{ "instance1" };
+  props["uniqueProp"] = instanceId;
+  configObject->Update(props);
```
Comment (pelliott-mathworks) [inline compendium/DeclarativeServices/test/TestDSConfigAdminOperationOrder.cpp:64]: This Update happens so far before the Deactivate (stopping of dependent bundle) that I don't think it's a valid test. The Update would complete well before you even try to Deactivate the dependent component. I think you should restructure this test so this Update is immediately followed by depBundle.stop();

**Review by pelliott-mathworks — CHANGES_REQUESTED** [review]

**Review by pelliott-mathworks — APPROVED** [review]
Looks good!

**Pull request merged**